sympy/calculus/tests/test_finite_diff.py

```diff
@@ -142,3 +142,18 @@ def test_differentiate_finite():
     assert (res2 - ref2).simplify() == 0
     raises(ValueError, lambda: differentiate_finite(f(x)*g(x), x,
                                                     pints=[x-1, x+1]))
+
+    res3 = differentiate_finite(f(x)*g(x).diff(x), x)
+    ref3 = (-g(x) + g(x + 1))*f(x + S.Half) - (g(x) - g(x - 1))*f(x - S.Half)
+    assert (res3 - ref3).simplify() == 0
```

Comment:
This should just be `assert res3 == ref3` (same for others below)

We generally prefer not to use simplify in test output and it isn't needed in this case.